idea/src/org/jetbrains/kotlin/idea/intentions/ReplaceMathMethodsWithKotlinNativeMethodsIntention.kt

```diff
@@ -18,18 +18,17 @@ package org.jetbrains.kotlin.idea.intentions
 
 import com.intellij.openapi.editor.Editor
 import org.jetbrains.kotlin.idea.core.replaced
-import org.jetbrains.kotlin.psi.KtCallExpression
-import org.jetbrains.kotlin.psi.KtDotQualifiedExpression
-import org.jetbrains.kotlin.psi.KtPsiFactory
-import org.jetbrains.kotlin.psi.createExpressionByPattern
+import org.jetbrains.kotlin.psi.*
 import org.jetbrains.kotlin.psi.psiUtil.getStrictParentOfType
 
 abstract class ReplaceMathMethodsWithKotlinNativeMethodsIntention(
-        text: String, val replacedMethodName: String, val mathMethodName: String
+        text: String, private val replacedMethodName: String, private val mathMethodName: String
 ) : SelfTargetingOffsetIndependentIntention<KtCallExpression>(KtCallExpression::class.java, text) {
 
     override fun applyTo(element: KtCallExpression, editor: Editor?) {
-        val target = element.getStrictParentOfType<KtDotQualifiedExpression>() ?: element
+        val target = element.getStrictParentOfType<KtDotQualifiedExpression>()?.takeIf {
+            (it.receiverExpression as? KtNameReferenceExpression)?.getReferencedName() == "Math"
+        } ?: element
```

Comment:
It's also not the best way, you can meet something like

```
Math.max(max(1, 3), max(2, 4))
```

and will have the same problem. Will fix it myself.